src/compiler/transformers/classFields.ts

```diff
@@ -633,6 +725,15 @@ namespace ts {
             //      HasLexicalDeclaration (N) : Determines if the argument identifier has a binding in this environment record that was created using
             //                                  a lexical declaration such as a LexicalDeclaration or a ClassDeclaration.
             const staticProperties = getProperties(node, /*requireInitializer*/ true, /*isStatic*/ true);

```

Comment:
I think I figured what I was trying to express in our meeting today 😅:

- Stop filtering out uninitialized properties here (`requireInitializer = false`).
- Remove `staticUninitializedPrivateProperties` logic below.
- Early return in `transformProperty` for public static uninitialized properties so that they are not transformed. We do that for private uninitialized properties already.
- `createPrivateStaticFieldInitializer` can already handle uninitialized properties.

Does this make sense? This would simply `visitClassExpression`, too.